### DISCUSSION TITLE
Strange route line after moving a waypoint

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#627] [Q&A] [answered]
If I load a gpx file and then modify it by moving waypoints, the original waypoint position is sometimes also retained and a direct connection to the next waypoint is created. In this example (created by BRouter, I took your preferred country): wp14 is moved to the top right on the bypass route and then suddenly there is a connection from its old position to wp15. The move of wp14 also changes something at wp13 - the screenshots:

Overview - with the new position of wp14
![1](https://github.com/user-attachments/assets/3b34bb3c-f02a-4de8-826b-9310b12502a4)

Details of wp14/15 and wp13 before the change
![2](https://github.com/user-attachments/assets/d499ef9f-a80f-4b6a-820f-3716f4c20bd7)

![3](https://github.com/user-attachments/assets/981f6793-277a-458a-a1f4-6fd442fd8077)

Details of wp14/15 and wp13 after the change
![5](https://github.com/user-attachments/assets/78b28a41-a259-4759-bc73-da638b0f46f8)

![6](https://github.com/user-attachments/assets/d9033b4a-7f4b-40ae-9439-ffa9e26f3d9b)

This error does not occur all the time, there must be corresponding circumstances. I only noticed it a few times when transferring bike tours from Komoot (planning) to Cruiser (navigation). Here are the example files from BRouter: 
[El Serrat (17,6 km).gpx.txt](https://github.com/user-attachments/files/16861181/El.Serrat.17.6.km.gpx.txt)
[El Serrat (17,6 km).geojson.txt](https://github.com/user-attachments/files/16861182/El.Serrat.17.6.km.geojson.txt)

**Comment by 0709wiwiwi** [comment]
Can it be caused by this ?https://github.com/devemux86/cruiser/discussions/204

No time to have a  test. Walking in thé Alps now 😀 Sunny weather expected.

**Reply by devemux86** [reply]
@0709wiwiwi this new feature is not yet publicly available in BRouter and Cruiser.

**Comment by devemux86** [comment]
@WolfgangZ42 thanks for the detailed report.

Can you post a complete route file with the waypoints (e.g. JSON format)?
(the geojson file does not contain the waypoints, must check them on export dialog)

Also post your app profile json file.

What have you selected in "Settings | Routing | Advanced | Route calculation"?

**Reply by WolfgangZ42** [reply]
In all cases of this problem the input was pure track (from Komoot) - nevertheless you get the waypoints as well:
[El Serrat (17,6 km).geojson.txt](https://github.com/user-attachments/files/16865126/El.Serrat.17.6.km.geojson.txt)
[El Serrat (17,6 km).gpx.txt](https://github.com/user-attachments/files/16865141/El.Serrat.17.6.km.gpx.txt)

and the profile: [Bike.json](https://github.com/user-attachments/files/16865020/Bike.json)

I've set Route calculation to "Intelligent calculation of route sections".

**Reply by devemux86** [reply]
I can reproduce it using the app profile file and importing the gpx.
App profile file is important to test using the same import options.

I will check what happens here.

**Comment by devemux86** [comment]
> I took your preferred country

I now work in Azores! :slightly_smiling_face:

**Reply by WolfgangZ42** [reply]
Is that an upgrade? 🤔 At least you may test the "avoid ferries" function. 😉

**Comment by devemux86** [comment]
@WolfgangZ42 thanks for this report. :+1: 

This is a complicated case and is related to the BRouter response.
I think I fixed it and it should work correctly in the next update.

**Comment by devemux86** [comment]
This should be fixed in the latest:
- https://github.com/devemux86/cruiser/discussions/610

**Comment by devemux86** [comment, marked as answer]
- Fixed in https://github.com/devemux86/cruiser/discussions/641
